packages/ios_platform_images/lib/ios_platform_images.dart

```diff
@@ -138,6 +138,89 @@ class IosPlatformImages {
     return _FutureMemoryImage(bytesCompleter.future, scaleCompleter.future);
   }
 
+  /// Loads an image from the system.  The equivalent would be:
+  /// `[UIImage systemImageNamed:name]`.
+  ///
+  /// Throws an exception if the image can't be found.
+  ///
+  /// The [name] is the symbol name as defined by iOS. You can browse a list of
+  /// symbols in the SF Symbols app (see below).
+  ///
+  /// The [pointSize] is the size of the image in iOS font points.
+  ///
+  /// The [colors] variable is a list of colors to be applied to the corresponding layer
+  /// of the icon, assuming that layer exists. An icon can have up to 3 layers,
+  /// if a layer color is not specified it will take the color of the most recent
+  /// valid index. Defaults to an empty list (black).
+  ///
+  /// For more information see [https://developer.apple.com/documentation/uikit/uiimagesymbolconfiguration/3810054-configurationwithpalettecolors?language=objc]
+  ///
+  /// If true, [preferMulticolor] overrides [colors] and asks iOS to provide its
+  /// preset multicolor variant of the symbol. Depending on the symbol, these colors
+  /// *may not be mutable*. To find out, use the SF Symbols app found at
+  /// [https://developer.apple.com/sf-symbols/].
+  /// Also note that regardless of mutability, the symbols default colors are
+  /// affected by the devices theme preference.
+  ///
+  /// The [weight] is the weight or thickness of the icon. The default is [FontWeight.normal].
+  ///
+  /// Multi-layer and multi-color symbols only work on iOS 15.0 and above.
+  /// **This method requires at least iOS 13.0.**
+  ///
+  /// See [https://developer.apple.com/documentation/uikit/uiimage/configuring_and_displaying_symbol_images_in_your_ui?language=objc]
+  static ImageProvider loadSystemImage(
+    String name,
+    double size, {
+    List<Color> colors = const <Color>[],
+    bool preferMulticolor = false,
+    ui.FontWeight weight = FontWeight.normal,
+  }) {
+    if (preferMulticolor) {
+      if (colors != const <Color>[]) {
+        throw ArgumentError(
+          'preferMulticolor and colors cannot be used together.',
+        );
+      }
+    }
+
+    final List<double> colorsRGBA = colors
+        .expand((Color color) => <double>[
+              color.red.toDouble() / 255,
+              color.green.toDouble() / 255,
+              color.blue.toDouble() / 255,
+              color.alpha.toDouble() / 255,
+            ])
+        .toList();
+
+    final Future<Map<String, Object?>?> loadInfo =
+        _channel.invokeMapMethod<String, Object?>(
+      'loadSystemImage',
+      <Object?>[
+        name,
+        size,
+        weight.index,
+        colorsRGBA,
+        preferMulticolor,
+      ],
```

Comment:
Doing all of this by index is pretty ugly to track across the language boundary. Perhaps this would be a good time to introduce Pigeon into this plugin.

(Failing that, we usually do this with a map so there are named keys.)

Reply:
@stuartmorgan I'm trying to figure out what a good Pigeon implementation would look like here, are you suggesting I move the Obj-C logic with the switch statement for indices into a Pigeon API? What would I pass through the platform channel in that case?

Reply:
I'm not sure what you mean exactly. With Pigeon you would declare a method with arguments; there wouldn't be indexes at all.

Reply:
I don't know what you mean by "another method". I'm saying that index-based arguments are very error-prone and hard to maintain, and that you should either:
- use a dictionary for the arguments, where at least there's no chance of accidentally extracting completely the wrong argument on the native side, or
- use Pigeon for this new API, since Pigeon manages the internals of passing parameters for you and totally eliminates this problem.

Reply:
Ah! I see now, I thought I would be using Pigeon in addition to platform channels, but I think I see now that you mean I should replace this platform channel with a Pigeon API that can handle my enum. Thank you for clarifying!